File: mobile-native-backup/ios/LocalStore/EncryptedStore.swift
import Foundation

protocol EncryptedStore {
    func saveCiphertextMessage(id: Int64, conversationID: Int64, senderID: Int64, receiverID: Int64, ciphertextB64: String, headerJSON: String, createdAt: Date) throws
    func lastSyncTimestamp() throws -> Date?
    func setLastSyncTimestamp(_ date: Date) throws
}

File: mobile-native-backup/android/app/src/main/java/com/securemsg/storage/EncryptedStore.kt
package com.securemsg.storage

interface EncryptedStore {
    suspend fun saveCiphertextMessage(
        id: Long,
        conversationId: Long,
        senderId: Long,
        receiverId: Long,
        ciphertextB64: String,
        headerJson: String,
        createdAt: String
    )

    suspend fun getLastSyncTimestamp(): String?
    suspend fun setLastSyncTimestamp(isoTimestamp: String)
}
